### PR TITLE
docs: add recommneded settings in overview

### PR DESCRIPTION
**Pull request by patatman**
Every infrastructure is different, and we can't enforce DNS endpoints
but we can recommend it.

Signed-off-by: Patatman <git@jeursen.nl>

# Pull Request

## Acceptance

Please use the following checklist:

- [ ] you linked an issue (if applicable)
- [ ] you included tests (if applicable)
- [x] you ran conformance (`make conformance`)
- [ ] you formatted your code (`make fmt`)
- [ ] you linted your code (`make lint`)
- [ ] you generated documentation (`make docs`)
- [ ] you ran unit-tests (`make unit-tests`)

> See `make help` for a description of the available targets.

---
This change is [Reviewable](https://reviewable.io/reviews/talos-systems/talos/2403)

### CODE REVIEW

File: docs/website/content/v0.6/en/configuration/overview.md [hunk 1]
```diff
@@ -14,3 +14,47 @@ Talos enforces a high level of security by using mutual TLS for authentication a
...
+``` yaml
+.....
+cluster:
+.....
+  apiServer:
+    image: k8s.gcr.io/kube-apiserver:v1.19.0-beta.1
+    certSANs:
+    - endpoint.example.local
+.....
+```
```
Comment (andrewrynhard) [inline docs/website/content/v0.6/en/configuration/overview.md:54]: We actually add this automatically. We can remove this part.

File: docs/website/content/v0.6/en/configuration/overview.md [hunk 2]
```diff
@@ -14,3 +14,47 @@ Talos enforces a high level of security by using mutual TLS for authentication a
 We recommend that the configuration of Talos be performed by a cluster owner.
 A cluster owner should be a person of authority within an organization, perhaps a director, manager, or senior member of a team.
 They are responsible for storing the root CA, and distributing the PKI for authorized cluster administrators.
+
+### Recommended settings
+
+Talos runs great out of the box, but if you tweak some minor settings it will make your life
+a lot easier in the future.
```
Comment (andrewrynhard) [inline docs/website/content/v0.6/en/configuration/overview.md:21]: Linter is gonna hit you here. One sentence per line is required.

**Comment by rsmitty** [comment]
@patatman I went ahead and rebased this PR. Do you mind addressing @andrewrynhard's comment around the cert sans and we'll get this merged?

**Comment by smira** [comment]
updated this PR with feedback from @andrewrynhard

**Review by rsmitty — APPROVED** [review]

**Pull request merged**